src/state/usePasscodeAuth/usePasscodeAuth.ts

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+export function getPasscode() {
+  const match = window.location.search.match(/passcode=(.*)&?/);
+  const passcode = match ? match[1] : window.sessionStorage.getItem('passcode');
+  return passcode;
+}
+
+export function fetchToken(name: string, room: string, passcode: string) {
+  return fetch(`/token`, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify({ user_identity: name, room_name: room, passcode }),
+  });
+}
+
+export function verifyPasscode(passcode: string) {
+  return fetchToken('verification name', 'verification room', passcode).then(async res => {
```

Comment:
Are `'verification name', 'verification room'` temporary identity and room values? If so, what do you think about renaming to `temp-identity` and `temp-room`?

Reply:
Sounds good. :+1: They are temporary.